logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageMapper.kt

```diff
@@ -176,53 +175,53 @@ class MessageMapperImpl(
     @Suppress("ComplexMethod")
     override fun fromMessageToLocalNotificationMessage(
         message: NotificationMessageEntity
-    ): LocalNotificationMessage =
-        when (val content = message.content) {
-            is MessagePreviewEntityContent.Text -> LocalNotificationMessage.Text(
-                LocalNotificationMessageAuthor(
-                    content.senderName ?: "",
-                    null
-                ), message.date, content.messageBody
+    ): LocalNotificationMessage? {
+        val sender = LocalNotificationMessageAuthor(
+            message.senderName.orEmpty(),
+            message.senderImage?.toModel()
+        )
+        return when (message.contentType) {
+            MessageEntity.ContentType.TEXT -> LocalNotificationMessage.Text(
+                author = sender,
+                text = message.text.orEmpty(),
+                time = message.date,
             )
 
-            is MessagePreviewEntityContent.Asset -> {
-                val type = if (content.type == IMAGE) LocalNotificationCommentType.PICTURE
-                else LocalNotificationCommentType.FILE
-                LocalNotificationMessage.Comment(LocalNotificationMessageAuthor(content.senderName ?: "", null), message.date, type)
+            MessageEntity.ContentType.ASSET -> {
+                val type = message.assetMimeType?.contains("image/")?.let {
+                    if (it) LocalNotificationCommentType.PICTURE else LocalNotificationCommentType.FILE
+                } ?: LocalNotificationCommentType.FILE
+
+                LocalNotificationMessage.Comment(sender, message.date, type)
             }
 
-            is MessagePreviewEntityContent.MissedCall ->
+            MessageEntity.ContentType.KNOCK -> {
```

Comment:
Revert this change 😅 and remove `LocalNotificationCommentType.KNOCK`

Reply:
Removing LocalNotificationCommentType.KNOCK may be problematic since the enum type is already in released apps and looking at the enum column adapter we see that is uses the enum name

```kotlin
class EnumColumnAdapter<T : Enum<T>> @PublishedApi internal constructor(
  private val enumValues: Array<out T>,
) : ColumnAdapter<T, String> {
  override fun decode(databaseValue: String): T = enumValues.first { it.name == databaseValue } // this line will throw NoSuchElementException if the database have a value that does not exists in the enum type

  override fun encode(value: T) = value.name
}
```